lego/connections.py

```diff
@@ -0,0 +1,134 @@
+        """Closes the connection at the exit from the context manager.
+
+        Args:
+            exc_type: Exception type.
+            exc_value: Exception value.
+            traceback: Exception traceback.
+        """
+        self.close()
+
+    @abc.abstractmethod
+    def close(self) -> None:
+        """Closes all connections."""
+        raise NotImplementedError()
+
+
+class SSHConnection(BaseConnection):
+    """SSH connection for remote machine.
+
+    This connection provides a shell, that can be used to run shell commands and upload or download files.
+
+    Usage example:
+    connection = SSHConnection('zebra', 'admin', 'root')
+    remote_shell = connection.shell
+    r_ls = shell["ls"]
+    remote_files = r_ls()
+
+    For more details on how to use plumbum.SshMachine see https://plumbum.readthedocs.io/en/latest/#user-guide
+    """
+
+    def __init__(self, hostname: str, username: str, password: str) -> None:
+        """Initiates SSH connection.
+
+        Args:
+            hostname: The hostname of the component we want to connect to.
+            username: Username for SSH connection.
+            password: Password for SSH connection.
+        """
+
+        # TODO: Check if Paramkio machine can be used with rpyc.DeployedServer.
+        #       SshMachine uses an ssh connection for every command, and paramkio use only one connection.
+        self._machine = plumbum.SshMachine(hostname, user=username, password=password)
+
+    @property
+    def shell(self) -> plumbum.SshMachine:
+        """The SSH connection to component."""
+
+        return self._machine
+
+    def close(self) -> None:
+        """Closes all SSH connections."""
+
+        self._machine.close()
+
+
+class RPyCConnection(SSHConnection):
+    """RPyC wrapper for SSH connection.
+
+    SSH connection to remote machine, provide the connection to remote RPyC SlaveService.
+    In case the machine doesn't already run SlaveService, it will be uploaded and run in a temporarily directory.
+    """
+
+    def __init__(self, hostname: str, username: str, password: str) -> None:
+        """Initiates SSH connection, and connects (and start if needed) to RPyC remote SlaveService.
+
+        Args:
+            hostname: The hostname of the component we want to connect to.
+            username: Username for SSH connection.
+            password: Password for SSH connection.
+        """
+        super().__init__(hostname, username, password)
+
+        self._server = None
+        try:
+            # Checks if the machine already runs RPyC SlaveService.
+            self._conn = rpyc.classic.connect(hostname, keepalive=True)
+        except ConnectionRefusedError:
+            # Upload RPyC and start SlaveService in a temporarily directory.
+            self._server = DeployedServer(self.shell)
+            self._conn = self._server.classic_connect()
+
+    @property
+    def rpyc_connection(self) -> rpyc.core.protocol.Connection:
```

Comment:
Why it is not simply named connection?
```suggestion
    def connection(self) -> rpyc.core.protocol.Connection:
```